external/avro/src/main/scala/org/apache/spark/sql/avro/functions.scala

```diff
@@ -65,6 +68,31 @@ object functions {
     new Column(AvroDataToCatalyst(data.expr, jsonFormatSchema, options.asScala.toMap))
   }
 
+  /**
+   * Converts a binary column of Avro format into its corresponding catalyst value.
+   * The specified subject must match actual schema of the read data, otherwise the behavior
+   * is undefined: it may fail or return arbitrary result.
+   * To deserialize the data with a compatible and evolved schema, the expected Avro schema can be
+   * set via the option avroSchema.
+   *
+   * @param data the binary column.
+   * @param subject the subject name in the schema-registry. eg. topic: t, key: t-key value: t-value
+   * @param schemaRegistryUri address of the schema-registry url
+   *
+   * @since 3.0.0
+   */
+  @throws(classOf[java.io.IOException])
+  @Experimental
+  def from_avro(
+      data: Column,
+      subject: String,
+      schemaRegistryUri: String): Column = {
```

Comment:
I would avoid adding this in an API. Can we combine both `avroSchema` and `avroSchemaUrl`? You can try `Schema.Parser().parse` first and falls back to try parsing URL if it fails. In that way, we wouldn't have to add another API.

Reply:
what are your reasons for avoiding it? In my opinion, they are different things, and it's not really complicated to maintain this new API. Also, there are other things that as I mentioned Databricks already implemented this API, so do you agree it is better to keep these same? it would provide a better user experience for developers.
Anyway, you are the boss here, I follow you. What's your idea?

Reply:
The reason for avoiding is to minimize the number of APIs to the end users, and better usability. Arguably using URL is not very common usage. Adding an option is fine but I would like to avoid adding an API dedicated for it.

This is Apache Spark, not the fork of other companies. I wasn't involved in that API design in Databricks. No one is a boss in Apache projects. I am suggesting and reviewing it in my own perspective. It's up to you to reject and wait for other committers' opinion, and let them merge unless somebody explicitly objects.

What's the reason of avoiding it in one option? Isn't it simpler and easier to reason?

cc @gengliangwang too.

Reply:
Take it, easy man. yes, your reasons make sense about minimizing the number of APIs. I'm going to adjust my PR.
thank you

Reply:
That's fine but maybe let's wait for @gengliangwang's opinion too. He's more an expert in this area, and might have a different idea.

Reply:
Cool, Okay.

Reply:
@HyukjinKwon Thanks for the ping. 
I am fine with either way.  
For the current design, I think we need to add another API with a field "options: java.util.Map[String, String]" as the other `from_avro` API do.

Reply:
@gengliangwang So this new API is fine and I have to write unittest?

Reply:
@blcksrx From my side, yes.

Reply:
Okay, I am fine with it too. I would follow @gengliangwang's judgement in this area.